src/components/StoriesGroup/StoriesGroup.tsx

```diff
@@ -104,12 +131,30 @@ export const StoriesGroup = ({
 
     const onGroupSelect = React.useCallback(
         (newGroupIndex: number) => {
-            setStoryIndex([newGroupIndex, 0]);
+            setStoryIndex({groupIndex: newGroupIndex, itemIndex: 0});
             onItemSelect?.([newGroupIndex, 0], true);
         },
         [onItemSelect],
     );
 
+    const externalGroupIndex = index.groupIndex;
+    const externalItemIndex = index.itemIndex;
+    React.useEffect(() => {
+        setStoryIndex((prevState) => {
+            if (
+                prevState.groupIndex === externalGroupIndex &&
+                prevState.itemIndex === externalItemIndex
+            ) {
+                return prevState;
+            }
```

Comment:
Do we need this check inside `setStoryIndexCallback`? Seems like we can do it before in `useEffect`

Reply:
[330eea3](https://github.com/gravity-ui/uikit/pull/464/commits/330eea319b3280ba670dd77a39652fd84796f647)